src/esm_runscripts/filedicts.py

```diff
@@ -341,6 +357,46 @@ def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
         else:
             raise Exception(f"Cannot identify the path's type of {path}")
 
+    def _check_file_info_is_complete(self):
+        error_text = ""
+        missing_vars = ""
+        types_text = ", ".join(self.all_model_filetypes)
+
+        if "type" not in self.keys():
+            error_text = (
+                f"{error_text}"
+                f"- the ``type`` variable is missing. Please define a ``type`` "
+                f"({types_text})\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``type``: forcing/input/restart/outdata/...\n"
+            )
+        else:
+            if self["type"] not in self.all_model_filetypes:
+                error_text = (
+                    f"{error_text}"
+                    f"- ``{self['type']}`` is not a supported ``type`` "
+                    f"(``files.{self.name}.type``), please choose one of the following "
+                    f"types: {types_text}\n"
+                )
+                self._original_filedict["type"] = f"``{self._original_filedict['type']}``"
+
+        missing_vars = (
+            f"Please, complete/correct the following vars for your file:\n\n"
+            f"{self.pretty_original_filedict}"
+            f"{missing_vars}"
+        )
+        if error_text:
+            error_text = (
+                f"The file dictionary ``{self.name}`` is missing relevant information "
+                f"or is incorrect:\n{error_text}"
+            )
+            user_error("File Dictionaries", f"{error_text}\n{missing_vars}")
```

Comment:
This is a requested change: is there a way to check **something** and either use user_error or raise an exception here?

The reason: if you ever want to use this class in your own script, maybe outside of the esm-tools context, the way it is now closely connects and couples it to esm-tools. There are (many) other parts of the code that do this too, that's a generic thing for the entire project, not just this sprint. It's often impossible to just import a class out of esm tools and use it without the whole, emm I guess "bloat"

Reply:
Maybe we can create a `file_error` routine that wraps `user_error`? I think it is a bit of an over-complication, after all this is built to be used in the context of ESM-Tools file structure and it already needs a lot of variables from the config.